tensorflow/compiler/xla/service/gpu/cudnn_fused_conv_rewriter.cc

```diff
@@ -297,6 +298,239 @@ StatusOr<bool> FuseConvAlpha(HloComputation* comp) {
+        graph_string.ChangeDataType(op->shape().element_type());
+        return CaptureConvGraphRecursive(user, operands, graph_string,
+                                         visited_instrs, final_instr, 0);
+      }
+    }
+
+    // If none of the matches was successful and the pattern level is below the
+    // maximum level, attempt to match at higher level.
+    if (pattern_level < max_pattern_level) {
+      return CaptureConvGraphRecursive(user, operands, graph_string,
+                                       visited_instrs, final_instr,
+                                       pattern_level + 1);
+    }
+  }
+
+  // The first entry in the serialized graph is the convolution. The size is
+  // greater than one if at least one match was succesful.
+  if (graph_string.Size() > 1) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+// Captures the graph of pointwise operations operating on the convolution.
+bool CaptureConvGraph(HloInstruction* instr,
+                      std::vector<HloInstruction*>& operands,
+                      GraphString& graph_string, HloInstruction*& final_instr,
+                      HloInstruction* x_scale, HloInstruction* w_scale,
+                      bool x_mult_scale, bool w_mult_scale) {
+  absl::flat_hash_set<int> visited_instrs;
+  graph_string.AppendOp("conv", instr->shape().element_type());
+
+  // Shift the scaling of the inputs to the output of the convolution.
+  if (x_scale && w_scale && x_mult_scale == w_mult_scale) {
+    HloInstruction* product =
+        instr->AddInstruction(HloInstruction::CreateBinary(
+            x_scale->shape(), HloOpcode::kMultiply, x_scale, w_scale));
+    operands.push_back(product);
+    graph_string.AppendOp(x_mult_scale ? "scale" : "invscale",
+                          instr->shape().element_type());
+  } else {
+    if (x_scale) {
+      operands.push_back(x_scale);
+      graph_string.AppendOp(x_mult_scale ? "scale" : "invscale",
+                            instr->shape().element_type());
+    }
+    if (w_scale) {
+      operands.push_back(w_scale);
+      graph_string.AppendOp(w_mult_scale ? "scale" : "invscale",
+                            instr->shape().element_type());
+    }
+  }
+  return CaptureConvGraphRecursive(instr, operands, graph_string,
+                                   visited_instrs, final_instr);
+}
+
+// Matches convolutions operating on FP8 inputs and filters and rewrites into a
+// ForwardGraph Custom Call. For scaled FP8 convolutions on Hopper systems, the
+// following steps are elided and rewritten into a ForwardGraph Custom Call:
+//
+// 1. Cast the filter and input from FP8 to a wider type such as FP16 or FP32.
+// 2. Unscale the filter and input by multiplying them by the corresponding
+// input scale.
+// 3. Evaluate the convolution based on the scaled filter and input.
+// 4. Optionally add a matrix bias to the result and apply a ReLU activation.
+// 5. Scale the output by dividing the output by the output scale.
```

Comment:
You allow the relu, matrix bias, and scaling to be done in any order, and done any number of times. I would combine these steps as "Do a series of elementwise transformations, where a transformation can be adding a matrix bias, applying a ReLU activation, or scaling the output by dividing by a scalar output scale.